corgidrp/mocks.py

```diff
@@ -28,6 +30,45 @@ def create_dark_calib_files(filedir=None, numfiles=10):
     dataset = data.Dataset(frames)
     return dataset
 
+def create_nonlinear_dataset(filedir=None, numfiles=2,em_gain=2000):
+    """
+    Create simulated data to non-linear data to test non-linearity correction.
+    
+    Args:
+        filedir (str): (Optional) Full path to directory to save to.
+        numfiles (int): Number of files in dataset.  Defaults to 2 (not creating the cal here, just testing the function)
+        em_gain (int): The EM gain to use for the simulated data.  Defaults to 2000.
+        
+    Returns:
+        corgidrp.data.Dataset:
+            The simulated dataset
+    """
+    filepattern = "simcal_nonlin_{0:04d}.fits"
+    frames = []
+    for i in range(numfiles):
+        prihdr, exthdr = create_default_headers()
+        #Add the EMGAIN to the headers
+        exthdr['EMGAIN'] = em_gain
```

Comment:
What/who is responsible for this keyword to be populated?

Reply:
EM Gain is an interesting one with a few different cases: 
1) If the commanded emgain is 1 we probably don't need to do anything. I'm looking in to the header keyword we might expect. There's one that is CMDGAIN that is the commanded gain, but I don't know if that's the K-gain or the emgain at the moment (I've asked Alex). 
2) If the EM Gain is v high we can measure it from the pre-scan region. 
3) If the EM gain is lower then we may not be able to do that, and I believe that it's an outstanding task to figure out how to do this. 

I suggest we have a step function that we call no matter what. If CMDGAIN =1 (assuming this is the correct keyword) then it just sets EMGAIN=1, otherwise it does a calculation and sets the keyword. In all cases its that same function that is responsible for setting that keyword. Calculating EMGAIN is a required step that doesn't have a stub yet, only because I based the stubs off the ttr5_analog.py script which wasn't based on the EMGAIN workflow. It should be done. 

Thoughts?

Reply:
Ok, just heard from Alex, this is on the list of nebulous stuff that we should be able to get into the primary header. I've added a requested Header keyword to Data Formats page on Confluence, CMDEMGN, for ComManDed EM GaiN.

Reply:
It looks like `CMDGAIN` is already a planned keyword in the extension header?

Ok understood. So, we'll need an additional pipeline step before this on that calculates the EMGAIN then.

Reply:
CMDGAIN is not the EMGAIN (confirmed by Alex)

But yes, we'll need that step.